tests/fido_test.py

```diff
@@ -35,47 +35,59 @@ def server_url():
     logging.basicConfig()
     logging.getLogger('twisted').setLevel(logging.CRITICAL)
 
-    class MyHandler(BaseHTTPServer.BaseHTTPRequestHandler):
+    class TestHandler(BaseHTTPServer.BaseHTTPRequestHandler):
         def echo(self):
-            if '/slow' in self.path:
+            if 'slow' in self.path:
                 time.sleep(SERVER_OVERHEAD_TIME)
+
             self.send_response(200)
-            self.send_header("Content-type", "application/json")
+
+            for k, v in self.headers.items():
+                self.send_header(k, v)
             self.end_headers()
+
             content_length = int(self.headers.get('Content-Length', 0))
-            self.wfile.write(json.dumps({
-                'headers': dict(self.headers),
-                'method': self.command,
-                'body': self.rfile.read(content_length)
-            }))
+            if content_length > 0:
+                self.wfile.write(self.rfile.read(content_length))
+
+        def content_length(self):
+            """Send back the content-length number as the response."""
+            self.send_response(200)
+
+            response = encode_to_bytes(self.headers.get('Content-Length'))
+            content_length = len(response)
+            self.send_header('Content-Length', content_length)
+            self.end_headers()
+
+            self.wfile.write(response)
 
         def do_GET(self):
-            self.echo()
+            if ECHO_URL in self.path:
+                self.echo()
 
         def do_POST(self):
-            self.echo()
+            if 'content_length' in self.path:
+                self.content_length()
+            elif ECHO_URL in self.path:
+                self.echo()
 
     class MultiThreadedHTTPServer(SocketServer.ThreadingMixIn,
                                   BaseHTTPServer.HTTPServer):
         request_queue_size = 1000
 
-    httpd = MultiThreadedHTTPServer(('localhost', 0), MyHandler)
+    httpd = MultiThreadedHTTPServer(('localhost', 0), TestHandler)
     httpd_thread = threading.Thread(target=httpd.serve_forever)
     httpd_thread.start()
     yield 'http://%s:%d/' % (httpd.server_address[0], httpd.server_address[1])
     httpd.server_close()
     httpd_thread.join()
 
 
-def test_unicode_url():
-    assert _url_to_utf8(u'繁') == '\xe7\xb9\x81'
-
-
 def test_fetch_basic(server_url):
-    response = fido.fetch(server_url).wait()
-    assert 'Content-Type' in response.headers
+    response = fido.fetch(server_url + ECHO_URL).wait()
+    assert response.headers.get(b'User-Agent') == [b'Fido/3.1.0']
```

Comment:
you might want to construct this with the version from __about__.__version__ not to have to change this all the time